app/demo-geant-integration/SensitiveDetector.cc

```diff
@@ -41,9 +54,29 @@ G4bool SensitiveDetector::ProcessHits(G4Step* step, G4TouchableHistory*)
         return false;
     }
 
-    auto time = step->GetPreStepPoint()->GetGlobalTime();
+    // Fill the hit data for this step
+    auto         touchable = step->GetPreStepPoint()->GetTouchable();
+    unsigned int id        = touchable->GetVolume()->GetCopyNo();
+    HitData      data{id,
+                 edep,
+                 step->GetPreStepPoint()->GetGlobalTime(),
+                 touchable->GetTranslation()};
+
+    // Add energy deposition for this cell if it was hit before
+    for (auto hit : *(collection_->GetVector()))
+    {
+        if (id == hit->data().id)
+        {
+            hit->add_edep(edep);
+            return true;
+        }
+    }
+
+    // Otherwise, create a new hit
+    collection_->insert(new SensitiveHit(data));
```

Comment:
How many `SensitiveHit` will you have to make per SD (or per event)? I think this `new` was "cheap" in the `acc01` demo because it defined a custom new/delete using G4Allocator, right? If so , should we do the same here?

Reply:
Okay, I added the operator overload.

Reply:
Sorry that I missed the question part.  The total number of sensitive hits will be equivalent to the number of steps inside sensitive volumes, so can be large, for example, with the CMS-like calorimeter if we do not merge hits inside the same readout cell.